### ISSUE TITLE
Invalid LUA results in invalid SHA

### ISSUE THREAD

**Issue by tlhunter** [enobufs/lured#1]
When a Redis script contains an invalid script (e.g. the text `INVALID LUA`) the resulting scripts object will resemble the following:

``` json
{
  "script": "INVALID LUA",
  "sha": "ERR Error compiling script (new function): user_script:1: '=' expected near 'INVALID'"
}
```

When this happens the sha should probably remain undefined similar to other error situations.

**Comment by enobufs** [comment]
I agree, but I found how 'SCRIPT LOAD' reacts to the case isn't ideal:

``` js
$ node
> var c = require('redis').createClient()
undefined
> c.multi([['script', 'load', 'INVALID LUA']]).exec(function () { console.log('args=', arguments); });
true
> args= { '0': null,
  '1': [ 'ERR Error compiling script (new function): user_script:1: \'=\' expected near \'LUA\' ' ] }
```

I does not report via the first arg (`err`) of the callback  :p
This may be a known bug reported elsewhere... I will investigate it.

**Comment by enobufs** [comment]
I have filed an issue on node-redis's github repo: https://github.com/mranney/node_redis/issues/715
will see how they respond. In the mean time, I will try to find a way to mitigate the situation. (validate the result, etc).

**Comment by tlhunter** [comment]
Could do `sha.indexOf('ERR') === 0`

**Comment by enobufs** [comment]
I like that. 'R' will never be in ascii representation of sha1 value.
The fixed landed on v0.0.3
